core/admin/admin.php

```diff
@@ -624,6 +624,21 @@ public function init_new_template() {
 		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_new_template_scripts' ] );
 	}
 
+	public function version_update_warning( $current_version, $new_version ) {
+		$current_version_minor_part = explode( '.', $current_version )[1];
+		$new_version_minor_part = explode( '.', $new_version )[1];
+
+		if ( $current_version_minor_part === $new_version_minor_part ) {
```

Comment:
What if it's an upgrade from 2.0.0 to 3.0.0? consider to unset $parts[2] from both and compare the arrays.